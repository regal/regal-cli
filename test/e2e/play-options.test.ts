import { spawn } from "child_process";
import { execRegal } from "../test-utils";

const testOut = (arg: string[], done) => {
    const bundlePath = "./test/e2e/resources/sample-game/temp/bundle.regal.js";

    const proc = spawn(execRegal, ["play", bundlePath].concat(arg), {
        cwd: process.cwd(),
        shell: true
    });

    let firstResponse = true;
    const commands = ["hey there!", "foo", "asdfjf", ":quit"];

    let OUTPUT = "";

    proc.stderr.on("data", data => {
        throw new Error("StdErr: " + data.toString());
    });

    proc.on("error", err => {
        throw new Error("ERROR: " + JSON.stringify(err));
    });

    proc.stdout.on("data", data => {
        OUTPUT += `${data.toString()}`;
        if (!firstResponse) {
            proc.stdin.write(commands.shift() + "\n");
        }
        firstResponse = false;
    });

    proc.on("exit", () => {
        expect(OUTPUT).toMatchSnapshot();
        done();
    });
};

describe("Play-Options", () => {
    it("Play-Options: --debug", done => testOut(["--debug"], done));

    it("Play-Options: --showMinor false", done =>
        testOut(["--showMinor", "false"], done));
});
