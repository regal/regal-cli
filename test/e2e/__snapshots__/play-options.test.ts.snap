// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play-Options Play-Options: --debug 1`] = `
"Now Playing: Sample Game by Lars
Type :quit to exit the game.

Hello, world!
Debug mode activated.

You just said hey there!.
What a cool word!
Prior command list: []

You've said 1 in total.
Here's a random string for you: sWCjwirFoT

You just said foo.
What a cool word!
Prior command list: [\\"hey there!\\"]
Recently, you said hey there!.
You've said 2 in total.
Here's a random string for you: dsOSJenvKp

You just said asdfjf.
What a cool word!
Prior command list: [\\"hey there!\\",\\"foo\\"]
Recently, you said foo and hey there!
You've said 3 in total.
Here's a random string for you: WCjwirFoTm
"
`;

exports[`Play-Options Play-Options: --showMinor false 1`] = `
"Now Playing: Sample Game by Lars
Type :quit to exit the game.

Hello, world!

You just said hey there!.

You've said 1 in total.
Here's a random string for you: apMNIdZZxr

You just said foo.
Recently, you said hey there!.
You've said 2 in total.
Here's a random string for you: HBROjdrGuZ

You just said asdfjf.
Recently, you said foo and hey there!
You've said 3 in total.
Here's a random string for you: pMNIdZZxrj
"
`;
